Package 020 updates

agnessnowplow opened the pull request:
**Overview**
1. Update wording / instructions for the upcoming release of the Attribution Modeling Accelerator. 
2. Replace sample data instructions for Snowflake
3. Add support for Bigquery and Databricks

**Details**
The original purpose was to add support for Databricks and BigQuery and reword instructions to fit the latest release (0.2.0) of the `snowplow-fractribution` dbt package but we have since then been receiving more feedback and further requests to update the accelerator so we can use this PR to update anything that can easily be changed for now.

**Pending points**
- banner to be replaced once we receive this from the design team - removed from now not to be a dependency

**Update topics considered so far**
- rewording `Fractribution`, where possible change it to `Attribution Modeling`
- updating instructions for v0.2.0 and remove support for v0.1.0 due to breaking changes
- rewording time estimates on the intro page
- rewording visualisation section and its aim
- replace the ecommerce plugin now that the new snowplow-ecommerce-plugin has become available
- refine modeling instructions around preparing transactional data (from ecommerce plugin)

Craig-Mace reviewed (commented):
Some suggestions and questions on some wording.

zloff reviewed (commented):
Looking good, my only concern is around the partial change in naming and ensuring it doesn't end up being more confusing - i.e. it is called attribution modeling here but fractribution in the dbt package. I think if this is the way forward we just have to ensure we make it clear what fractribution is for the user

agnessnowplow commented on content/modeling/modeling_3.md:
```diff
@@ -101,7 +142,7 @@ Head to the SQL editor of your choice (e.g.: Snowflake Web UI) to check the mode
+Take some time to familiarise yourself with the derived tables. These tables are used in the next step to attribute revenue to channels. Tables output by the snowplow_fractribution dbt package are:
 
 - `snowplow_fractribution_channel_counts`: Number of events grouped by channel, campaign, source and medium.
 - `snowplow_fractribution_channel_spend`: Spend on each channel, used in ROAS calculations.
```
I have reworded this to the same pattern as the description for the other tables above it, only including `fractrubution` in it's model output name.

agnessnowplow commented on content/_index.md:
```diff
@@ -66,20 +67,17 @@ You can use the provided sample data to follow steps 1-3. If you have your own w
 #### Prerequisites
```
I reworded and highlighted this sentence and added an additional description about this on the intro page of the tracking section.

agnessnowplow commented on content/modeling/modeling_4.md:
```diff
@@ -4,69 +4,96 @@ weight = 4
 post = ""
```
It is the name of the model output, I simplified it to report table to make it simple.

Craig-Mace replied:
Got it, makes sense, thanks Agnes

agnessnowplow commented on content/visualisation/_index.md:
```diff
@@ -21,4 +21,4 @@ flowchart LR
 {{</mermaid >}}
 
-This section provides an example of the fractibution report table and its graphical representation.
+This section provides an example of the fractibution report table and its graphical representation to give you a better understanding of the outcome of the analysis.
```
Same here as above.

agnessnowplow commented on content/_index.md:
```diff
@@ -8,36 +8,37 @@ weight = 1
 
-This accelerator will show you how to perform fractional attribution analysis on your Snowplow data, enabling you to attribute a portion of the value of a conversion to specific channels based on the conversion pathway. With this information, you can calculate the revenue per channel and Return on Advertising Spend (ROAS*) for each marketing channel, giving you a data-driven approach to making st ...
+This accelerator will show you how to perform attribution modeling on your Snowplow data, enabling you to attribute a portion of the value of a conversion to specific channels based on the conversion pathway. With this information, you can calculate the revenue per channel and Return on Advertising Spend (ROAS*) for each marketing channel, giving you a data-driven approach to making strategic mar ...
 
```
I agree completely and for this same reason I included this bit at the modeling bit: 
<img width="1234" alt="image" src="https://user-images.githubusercontent.com/95634439/223404481-fbd8abbd-3cf5-4d18-b781-5724c855d2df.png">
I will move this to the intro page to make it clear from the get-go.

agnessnowplow commented on content/modeling/modeling_3.md:
```diff
@@ -14,38 +14,77 @@ The snowplow_fractribution dbt package comes with a list of variables specified
+  snowplow__conversion_hosts: ['a.com']
+  snowplow__path_lookback_steps: 0
+  snowplow__path_lookback_days: 30
+  snowplow__path_transforms: {'exposure_path', null}
```
Good spot!!

agnessnowplow commented on content/modeling/modeling_3.md:
````diff
@@ -14,38 +14,77 @@ The snowplow_fractribution dbt package comes with a list of variables specified
+  snowplow__conversions_source: 'atomic.sample_events_fractribution'
 ```
 
 **Path Transform Options**
````
Indeed, and two are actually missing, amended this.

agnessnowplow commented on content/modeling/modelling_5.md:
```diff
@@ -20,6 +20,6 @@ You can then run fractribution using this column instead of a revenue column. An
 | 2022-06-03                | 2022-08-01              | Video             | 1.5         | 1.5     | 10000.0| 0.00014|
 
-Note that 'revenue' in this table is now being calculated based the form_submitted column in the events/conversions table, where a conversion will only ever have a value of 1. This is why the values for conversions and revenue are identical. 
+Note that 'revenue' in this table is now being calculated based the form_submitted column in the events/conversions table, where a conversion will only ever have a value of 1. This is why the values for conversions and revenue are identical.
```
Funny how our mind just adds this, never noticed it, haha.

agnessnowplow commented:
Updated this PR after a long pause waiting for the banner and the S3 buckets to be created to update the sample data methods and instructions. You can see the latest status updated in the PR's intro.

I rebased everything against master due to the other PRs that got merged hence the force push. I only touched the last three commits:

1. [Remove banner](https://github.com/snowplow/marketing-attribution-accelerator/pull/10/commits/5fbb7b0f1448280e849618d9e84f44f2288b3ec3) I removed the banner for now, no point in waiting any longer in this PR.
2. [Reword visualisation based on user feedback](https://github.com/snowplow/marketing-attribution-accelerator/pull/10/commits/c7fea50afd161554ab39b1ab3ba30e57a035bfee) 
3. [Update sample data upload instructions for all 3 warehouse targets](https://github.com/snowplow/marketing-attribution-accelerator/pull/10/commits/458d410140eb1e9854d918ffaacce50008525c7e) Removed the zip file, now everything should refer to the S3's csv (that needs fixing still tomorrow as I noticed the current file had issues). Snowflake will use a much more simplified approach using direct S3 transfer, BQ and Databricks will just use the usual csv upload method as for now it is simpler than having to authenticate on S3 (credentials not even created/provided yet).

Apart from these changes the only pending point is Peter's rework on the ecommerce tracking bits then it should be ready to merge.

miike commented on content/_index.md:
```diff
@@ -60,26 +59,31 @@ gantt
   - New dbt project created and configured
 - Python 3 Installed
-- Snowflake account and a user with access to create schemas and tables
+- Snowflake, Databricsks or BigQuery account and a user with access to create schemas and tables
```
sp - `Databricks`

miike commented on content/modeling/modeling_3.md:
```diff
@@ -14,56 +14,97 @@ The snowplow_fractribution dbt package comes with a list of variables specified
 **Configure the conversion_clause macro**
 
-The conversion_macro specifies how to filter Snowplow events to only conversion events. How this is filtered will depend on your definition of a conversion. The default is filtering to events where `tr_total > 0`, but this could instead filter on `event_name = 'checkout'`, for example. If you are using the e-commerce model, you will still need to set this for the fractribution code to run (even t ...
+The conversion_macro specifies how to filter Snowplow events to only conversion events. How this is filtered will depend on your definition of a conversion. The default is filtering to events where `tr_total > 0`, but this could instead filter on `event_name = 'checkout'`, for example. If you are using the e-commerce model, you will still need to set this to enable the models to run (even though  ...
```
This copy may need to be changed to reflect that it defaults to the ecomm package now rather than `tr_total`.

agnessnowplow replied:
The sample data still has `tr_total` as it is simpler to prepare and it is done before we talk about using the ecommerce package in the tracking steps but I added a sentence to clarify this! Also added a third option for the `conversions_source` where I describe that they don't have to do anything extra if they used the sample data just define that as the conversion source in their project.yml.

miike commented on content/upload/upload_1.md:
```diff
@@ -334,78 +323,435 @@ SELECT
+
+On the `Schema` section:
+
+- Click `Auto-detect`
```
We can load direct from S3 but this requires setting up a connection in BigQuery, so probably not worth it for this exercise.
